### ISSUE TITLE
Unjellyable.unjellyFor does not return self

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11110]
|<img alt="ml.flex's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| ml.flex reported|
|-|-|
|Trac ID|trac#1629|
|Type|defect|
|Created|2006-04-10 14:47:27Z|

Unjellyable.unjellyFor does not return self, unlike, for instance, RemoteCopy.unjellyFor. This means that the default implementation of Unjellyable is unjellied as None. 

See this example: it should print something like: 
<unjellybug.MyUnjellyable instance at 0x008F90D0>
But instead it prints None.
```
from twisted.spread import pb, jelly

from twisted.internet import reactor

class MyUnjellyable(jelly.Unjellyable):
    pass

class MyJellyable(jelly.Jellyable):
    pass

pb.setUnjellyableForClass(MyJellyable, MyUnjellyable)

class testRoot(pb.Root):
    def remote_echo(self, dat):
        print dat

def copydata(r):
    dat = MyJellyable()
    d = r.callRemote("echo", dat)
    d.addCallback(lambda x: reactor.stop())

if __name__ == '__main__':
    reactor.listenTCP(8789, pb.PBServerFactory(testRoot()))
    factory = pb.PBClientFactory()
    reactor.connectTCP("127.0.0.1", 8789, factory)
    factory.getRootObject().addCallback(copydata)
    reactor.run()

```

<details><summary>Searchable metadata</summary>

```
trac-id__1629 1629
type__defect defect
reporter__ml_flex ml.flex
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1144680447000000 1144680447000000
changetime__1150509798000000 1150509798000000
version__ 
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph commented|
|-|-|

This test needs to be expressed as a unit test in order to make this change.  I would recommend something simpler, though, that doesn't involve establishing a PB connection.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

Ready for review in [branches/unjellyable-1629](branches/unjellyable-1629)

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)|@jerub set owner to @glyph|
|-|-|

Thanks for the test, it's simpler than the example cited in the ticket, and makes sense.

unjellyFor doesn't have a docstring. Please add one. What is 'jellyList'? Why is it an indexable sequence with something at index 1?

Add the docstring, make sure it uses @param, and @return, and merge, thankyou.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @exarkun|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `closed`|
|-|-|

(In [17256]) Merge unjellyable-1629

Author: exarkun
Reviewer: jerub
Fixes [#11110](https://github.com/twisted/twisted/issues/11110)

This fixes the IUnjellyable interface definition to match up with reality,
improves one of its docstrings, adds test coverage for the very simple
implementation of it in twisted.spread.jelly.Unjellyable, and fixes that
implementation.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
